### PR TITLE
gh-101819: Clean up _io windows console io after gh-104197

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-101819

### CODE REVIEW

**Review by kumaraditya303 — APPROVED** [review]

File: Modules/_io/winconsoleio.c [hunk 1]
```diff
@@ -263,7 +263,7 @@ _io__WindowsConsoleIO___init___impl(winconsoleio *self, PyObject *nameobj,
     int fd_is_own = 0;
     HANDLE handle = NULL;
 
-#ifdef Py_DEBUG
+#ifdef NDEBUG
```
Comment (vstinner) [inline Modules/_io/winconsoleio.c:266]: Hum, that's wrong :-( You should only emit assertion if the NDEBUG macro is **not** defined... Double negation, I hate that. With the current code, in short you declare the state variable which is not used, but the assertion is always skipped!

**Review by vstinner — COMMENTED** [review]
Review after merge: the NDEBUG macro is misused.

File: Modules/_io/winconsoleio.c (hunk 1 above)
Comment (kumaraditya303) [reply Modules/_io/winconsoleio.c:266]: Oops, somehow I read it as `ifndef` and approved

Comment (erlend-aasland) [reply Modules/_io/winconsoleio.c:266]: Argh! Fix-up of the fix-up in flight.

Comment (erlend-aasland) [reply Modules/_io/winconsoleio.c:266]: Let's await Victor's review this time :)

Comment (vstinner) [reply Modules/_io/winconsoleio.c:266]: It's ok. It's easy to make this mistake. I always look at existing code and... I copy/paste it :-D

**Pull request merged**